File: src/test/java/no/nav/samordning/innlasting/ApplicationPropertiesTest.java
package no.nav.samordning.innlasting;

import org.junit.jupiter.api.Test;

import java.util.HashMap;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertEquals;

class ApplicationPropertiesTest {

    private static final String JDBC_URL = "jdbc_url";
    private static final String JDBC_URL_KEY = "DB_URL";

    private static Map<String, String> environment = new HashMap<>();

    @Test
    void getDbPropertyFromEnvironment_returns() {
        environment.put(JDBC_URL_KEY, JDBC_URL);
        String jdbcUrl = ApplicationProperties.getFromEnvironment(environment, JDBC_URL_KEY);

        assertEquals(JDBC_URL, jdbcUrl);
    }

    @Test
    void getDbPropertyFromEnvironment_throws_MissingDatabaseConfig_when_property_has_no_value_in_environment() {
        environment.put(JDBC_URL_KEY, null);
        assertThrows(MissingApplicationConfig.class,
                () -> ApplicationProperties.getFromEnvironment(environment, JDBC_URL_KEY)
        );
    }

    @Test
    void getDbPropertyFromEnvironment_throws_MissingDatabaseConfig_when_property_is_not_in_environment() {
        assertThrows(MissingApplicationConfig.class,
                () -> ApplicationProperties.getFromEnvironment(environment, null)
        );
    }

}

File: src/test/kotlin/no/nav/samordning/innlasting/ApplicationPropertiesTest.kt
package no.nav.samordning.innlasting

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test
import java.util.*

internal class ApplicationPropertiesTest {

    @Test
    fun getDbPropertyFromEnvironment_returns() {
        environment[JDBC_URL_KEY] = JDBC_URL
        val jdbcUrl = ApplicationProperties.getFromEnvironment(environment, JDBC_URL_KEY)

        assertEquals(JDBC_URL, jdbcUrl)
    }

    @Test
    fun getDbPropertyFromEnvironment_throws_MissingDatabaseConfig_when_property_has_no_value_in_environment() {
        @Suppress("UNCHECKED_CAST")
        environment[JDBC_URL_KEY] =  ({ null } as () -> String)()
        assertThrows(MissingApplicationConfig::class.java
        ) { ApplicationProperties.getFromEnvironment(environment, JDBC_URL_KEY) }
    }

    @Test
    fun getDbPropertyFromEnvironment_throws_MissingDatabaseConfig_when_property_is_not_in_environment() {
        assertThrows(MissingApplicationConfig::class.java
        ) { ApplicationProperties.getFromEnvironment(environment, "invalid") }
    }

    companion object {

        private const val JDBC_URL = "jdbc_url"
        private const val JDBC_URL_KEY = "DB_URL"

        private val environment = HashMap<String, String>()
    }

}
